MHArmory.Search/Incremental/IncrementalSolverData.cs

```diff
@@ -0,0 +1,391 @@
+                {
+                    continue;
+                }
+
+                bool currentBetter = IsBetter(equipment, other);
+                bool otherBetter = IsBetter(other, equipment);
+
+                if (currentBetter)
+                {
+                    if (!otherBetter)
+                    {
+                        other.IsSelected = false;
+                        //list.Remove(other);
+                        //i--;
+                    }
+                }
+                else if (otherBetter)
+                {
+                    equipment.IsSelected = false;
+                    return;
+                }
+            }
+        }
+
+        private bool IsBetter(ElectionModel model, ElectionModel other)
+        {
+            if (HasBetterByNonStats(model, other))
+            {
+                return true;
+            }
+
+            /*if(!HasBetterByNonStats(other, model))
+            {
+                if (HasBetterStats(model, other))
+                {
+                    return true;
+                }
+            }*/
+
+            return false;
+        }
+
+        private bool HasBetterByNonStats(ElectionModel model, ElectionModel other)
+        {
+            if (!model.FullSet && other.FullSet)
+            {
+                return true;
+            }
+
+            if (HasBetterSlotEfficiency(model, other))
+            {
+                return true;
+            }
+
+            /*if (HasBetterSlots(model, other))
+            {
+                return true;
+            }*/
+
+            if (HasBetterDesiredAbilities(model, other))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private bool HasBetterSlotEfficiency(ElectionModel model, ElectionModel other)
+        {
+            foreach (KeyValuePair<int, int> efficiencyPair in model.SkillEfficiencies)
+            {
+                int skillId = efficiencyPair.Key;
+                int efficiency = efficiencyPair.Value;
+                int otherEfficiency = other.SkillEfficiencies[skillId];
+                if (efficiency > otherEfficiency)
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        private bool HasBetterSlots(ElectionModel model, ElectionModel other)
+        {
+            if (model.SlotCount > other.SlotCount)
+            {
+                return true;
+            }
+
+            if (model.MaxSlot > other.MaxSlot)
+            {
+                return true;
+            }
+
+            if (model.SlotSum > other.SlotSum)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private bool HasBetterDesiredAbilities(ElectionModel model, ElectionModel other)
+        {
+            if (model.Equipment.Name["EN"].StartsWith("Nerg") && other.Equipment.Name["EN"].StartsWith("Nerg"))
```

Comment:
Looks like debug code ^^
Please remove it if it is, or add a very clear comment why you need that.